### DISCUSSION TITLE
format (on save) rust nightly support

### DISCUSSION THREAD

**Discussion by dcow** [zed-industries/zed#6506] [Feature Requests]
We use nightly features of `cargo fmt` so we have to run like `cargo +nightly fmt` otherwise it messes our formatting up. How can I configure zed to use nightly rust when running format (on save)?

**Comment by therealdannzor** [comment]
bump

**Comment by neongreen** [comment]
```
// .zed/settings.json
{
  "lsp": {
    "rust-analyzer": {
      "initialization_options": {
        "rustfmt": {
          "extraArgs": ["+nightly"]
        }
      }
    }
  }
}
```

**Reply by dream-dasher** [reply]
Can confirm that works.   Thanks! =)

(By contrast, trying something like in the current zed docs:
```json
    "Rust": {
        "formatter": {
         "external": {
           "command": "cargo",
           "arguments": ["+nightly", "fmt"]
         }
       }
    }
```

Does _not_ work -- as this is telling zed to feed the file contents into the command.  And cargo fmt doesn't accept a text pipe (so the result is the file contents are erased on format. :)

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
